lib/Corres_Method.thy

```diff
@@ -161,6 +161,10 @@ method corres
 
 section \<open>Corres rule setup\<close>
 
+(* Avoid using equations in the assumptions. subst_all gets around (no_asm_use) in some cases,
+   which we don't want. *)
+lemmas [corres_no_simp] = subst_all
```

Comment:
I don't understand this one. I also don't understand how it is added in this PR, but its use seems to already appear before this PR.

Reply:
I was confused by this as well. My understanding is that `subst_all` was causing simplification problems before this PR. The change here is to add it to the `corres_no_simp` set, which is then passed to `simp del` inside of the `corres` method (https://github.com/seL4/l4v/blob/555d1a49427171c7283be5a26bd369bef4d028d8/lib/Corres_Method.thy#L130C59)

Reply:
Ahh, so my brain blanked on this alternate syntax to `declare subst_all[corres_no_simp]`, thanks for the clarification!

Reply:
Yeah, it isn't a syntax that I've consciously used before. Based on the following examples in this file it looks like a more compact way to simultaneously give multiple lemmas the same attribute, although we would have to be careful that we aren't modifying any of them and putting unnamed lemmas into attribute sets.

Reply:
Mm. Yes, I think it's because we normally either see `[simp]` as the attribute, or we see multiple lemmas. Seeing `[corres_no_simp]` and only one lemma threw me off, but it's the more stable notation if we want to add more to the set in future. Regarding the modifying, I think `declare` would have the same problem. I guess you could remove it from the set using the same modified thm.

Reply:
I can confirm both the syntax and the intention discussed in this thread. I'll mark it as resolved, but if you want me to do anything to help clear up confusion, let me know.